adaptdl/adaptdl/torch/_metrics.py

```diff
@@ -111,12 +113,14 @@ def _fit_perf_params():
     num_nodes = np.array([key[0] for key in state.profile])
     num_replicas = np.array([key[1] for key in state.profile])
     local_bsz = np.array([key[2] for key in state.profile])
+    grad_acc_steps = np.array([key[3] for key in state.profile])
     values = state.profile.values()
     step_time = np.array([val["step_time"] / val["count"] for val in values])
     sync_time = np.array([val["sync_time"] / val["count"] for val in values])
     compute_time = step_time - sync_time
     state.perf_params = adaptdl.speedup.fit(
-            num_nodes, num_replicas, local_bsz, step_time, compute_time)
+        num_nodes, num_replicas, local_bsz, grad_acc_steps,
+        step_time, compute_time)
```

Comment:
I think there is a problem here. Based on how the metrics are collected, `step_time` contains the average step time across both accumulated and non-accumulated steps. However, the `SpeedupFunction` calculates the sum of step times over a single gradient accumulation cycle.

One suggestion is to collect different `step_time` for accumulated steps and non-accumulated steps. That makes it a bit clearer what's being measured/calculated at all times.

Reply:
Fixed